database-mongoose/reviews.service.js

```diff
@@ -0,0 +1,217 @@
+            $group:
+            {
+              _id: '$productId',
+              avgRating: { $avg: "$rating" },
+              totalRatings: { $sum: 1 }
+            }
+          }
+        ]
+      )
+
+    })
+    .then(ratingsByCustomer => {
+      console.log('ratingsByCustomer : ', ratingsByCustomer);
+      //calculate the customer rating count in % and save data in object to return
+      var reviewSummary = {
+        averageRating: ratingsByCustomer[0].avgRating.toFixed(2),
+        totalRatings: ratingsByCustomer[0].totalRatings,
+        fiveStar: (countByReviews.find(obj => obj._id === 5) ? ((countByReviews.find(obj => obj._id === 5).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        fourStar: (countByReviews.find(obj => obj._id === 4) ? ((countByReviews.find(obj => obj._id === 4).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        threeStar: (countByReviews.find(obj => obj._id === 3) ? ((countByReviews.find(obj => obj._id === 3).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        twoStar: (countByReviews.find(obj => obj._id === 2) ? ((countByReviews.find(obj => obj._id === 2).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        oneStar: (countByReviews.find(obj => obj._id === 1) ? ((countByReviews.find(obj => obj._id === 1).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%')
+      };
+      console.log(reviewSummary);
+      var promiseSummary = new Promise((resolve, reject) => {
+        resolve(reviewSummary)
+      });
+      return promiseSummary;
+
+    })
+    .catch(err => console.log('err :', err))
+};
+
+// /Reviews/getReviewsByFeature/:productId
+//get average value of the features
+let getReviewsByFeature = (productId) => {
+  return ReviewsModel.aggregate([
+    {
+      $match: { productId: { $eq: productId } }
+    },
+    {
+      $group: {
+        _id: '$productId',
+        easeToUseAvg: { $avg: "$easeToUse" },
+        voiceRecognitionAvg: { $avg: "$voiceRecognition" },
+        techSupport: { $avg: "$techSupport" },
+        valueForMoney: { $avg: "$valueForMoney" },
+        qualityOfMaterial: { $avg: "$qualityOfMaterial" },
+        batteryLife: { $avg: "$batteryLife" }
+
+      }
+    }
+  ])
+    .then(result => {
+      console.log('getReviewsByFeature: ', result[0])
+      var reviewsByFeature = {
+        easeToUse: result[0].easeToUseAvg.toFixed(1),
+        voiceRecognition: result[0].voiceRecognitionAvg.toFixed(1),
+        techSupport: result[0].techSupport.toFixed(1),
+        valueForMoney: result[0].valueForMoney.toFixed(1),
+        qualityOfMaterial: result[0].qualityOfMaterial.toFixed(1),
+        batteryLife: result[0].batteryLife.toFixed(1)
+
+      };
+
+      var reviewfeaturesPromise = new Promise((resolve, reject) => {
+        resolve(reviewsByFeature)
+      })
```

Comment:
Likewise here with the reject case.

Reply:
Made note of it. Planning to take care of it for next PR.